central/securedclustercertgen/certificates.go

```diff
@@ -41,6 +46,14 @@ func IssueLocalScannerCerts(namespace string, clusterID string) (*storage.TypedS
 	return certIssuer.issueCertificates(serviceTypes, namespace, clusterID)
 }
 
+// IssueSecuredClusterCerts issues certificates for the services of a secured clusters (excluding local scanner).
```

Comment:
Actually I am wondering, are there any downsides of *not* making this distinction between secured clusters with and without local scanning enabled?
I mean, how about we define a set of *all* possible certificates for a secured cluster and we unconditionally fetch and refresh all of them. Including optional things like scanner v4.
Wouldn't that make our life significantly simpler here? I am also thinking about situations where components are enabled/disabled post-install.

Reply:
What is @porridge's opinion on this? I have the feeling getting all (possibly useful) certs for secured clusters would simplify the code by allowing us to get rid of actually unnecessary branching. Are there any downsides I am missing?

I have a tendency towards such a change.

Reply:
It's true, it would allow us to simplify the logic in the Sensor side by quite a bit. (on Central we side we still need to have the old API for backwards compat)

So I would also be in favor of returning all the certificates.

In that case, I have one more question. We said that for `sensor`, `collector` and `admission-controller` we'd use new secret names, so that we don't interfere with the automation that some customers might have. But should we do that for scanner certs too? I would say no, i.e. we can use the same secrets, because Sensor was already managing them.

Reply:
The argument for using the existing names is that we wouldn't need any changes in scanner. But better to go for consistency.

Reply:
> So I would also be in favor of returning all the certificates.

That's what the current version of the PR does, right? I think it's fine. Let's make sure sensor is robust against seeing unexpected certs in response.

As for the second question, +1 to consistency.